### ISSUE TITLE
The SBOM issue persists despite the successful pipeline for "AI.Models.SmartPicker-Official" and "AI.Models.NL2PowerFX-Official."

### ISSUE THREAD

**Issue by hravellamicrosoft** [microsoft/component-detection#928]
I wanted to bring to your attention that I have identified similar issues to the ones reported in "https://github.com/microsoft/component-detection/issues/862 " for the "AI.Models.SmartPicker-Official", "AI.Flows.NextBestAction-Official" and "AI.Models.NL2PowerFX-Official" repositories regarding SBOM (Software Bill of Materials).

I have observed successful builds of SBOM manifest artefacts for the two pipelines. However, issues have arisen in the SBOM security compliance aspect.
https://dev.azure.com/powerbi/Business360%20AI/_build/results?buildId=8532158&view=logs&j=0a3c71cb-e72e-5356-5209-9a5ddb8da13b&t=ffa8cf66-1c16-5d05-86dd-8eb2d8103242
https://dev.azure.com/powerbi/Business360%20AI/_build/results?buildId=8604750&view=logs&j=0a3c71cb-e72e-5356-5209-9a5ddb8da13b&t=ffa8cf66-1c16-5d05-86dd-8eb2d8103242
https://dev.azure.com/powerbi/Business360%20AI/_build/results?buildId=8596917&view=logs&j=0a3c71cb-e72e-5356-5209-9a5ddb8da13b&t=ffa8cf66-1c16-5d05-86dd-8eb2d8103242

Could you please review and address these issues with immediate attention, as they have already exceeded the SLA?

Regards,
Hari

AB#2127636

**Comment by cobya** [comment]
Relevant stack trace as Component Detection is an open source project:
```##[error]Error while parsing pip components in "/mnt/vss/_work/1/b/2/_work/1/s/setup.py"
System.IO.InvalidDataException: Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.
   at System.IO.Compression.ZipArchive.ReadCentralDirectory()
   at System.IO.Compression.ZipArchive.GetEntry(String entryName)
   at Microsoft.ComponentDetection.Detectors.Pip.PyPiClient.FetchPackageDependenciesAsync(String name, String version, PythonProjectRelease release)
   at Microsoft.ComponentDetection.Detectors.Pip.PythonResolver.FetchPackageDependenciesAsync(PythonResolverState state, PipDependencySpecification spec)
   at Microsoft.ComponentDetection.Detectors.Pip.PythonResolver.ProcessQueueAsync(ISingleFileComponentRecorder singleFileComponentRecorder, PythonResolverState state)
   at Microsoft.ComponentDetection.Detectors.Pip.PythonResolver.ResolveRootsAsync(ISingleFileComponentRecorder singleFileComponentRecorder, IList`1 initialPackages)
   at Microsoft.ComponentDetection.Detectors.Pip.PipComponentDetector.OnFileFoundAsync(ProcessRequest processRequest, IDictionary`2 detectorArgs)```

**Comment by hravellamicrosoft** [comment]
is there any update on this issue ?

**Comment by hravellamicrosoft** [comment]
@cobya - Could you kindly provide an update on this issue, as it is currently blocking three other issues?

**Comment by cobya** [comment]
@hravellamicrosoft does this issue occur consistently or only once in a while? If it repros consistently, can you please share an example of your requirements.txt which is being scanned?

**Comment by hravellamicrosoft** [comment]
For the three projects 'AI.Models.SmartPicker-Official,' 'AI.Flows.NextBestAction-Official,' and 'AI.Models.SC.Copilot,' I can confirm that the issue seems to be occurring consistently. Could you please provide more details on what exactly you're seeking from these repositories?

**Comment by cobya** [comment]
Sure, if you could please provide the Python requirements.txt file which is being used to restore the dependencies that would help us diagnose the issue. You can share these with me via email / Teams if they are internal sensitive.

**Comment by hravellamicrosoft** [comment]
I have hared requirement.txt file via email on 4th Jan. I hope you get it, let me know if you still need info on this issue to make progress

**Comment by cobya** [comment]
This should be somewhat resolved with #962 but while investigating, I found that there is another issue which can be hit with oddly formed packages on PyPi. I'll be closing this issue since this shouldn't be happening anymore and opened #963 to dive deeper. That may also be handled via #980.
